### PR TITLE
Developer guide update

### LINKED ISSUE #171: Update DG

### PR DESCRIPTION
**Pull request by kimberlytmq**
Update manual test case appendix and effort appendix

Closes #171

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -541,20 +598,63 @@ testers are expected to do more *exploratory* testing.
...
+## **Appendix: Effort**   
+This project required substantial effort due to the complexity of expanding beyond the initial structure of AddressBook Level 3 (AB3).
+The following describes the challenges, difficulty level, and effort involved, as well as achievements attained through enhancements and
+additional features.
+
+### Difficulty Level and Challenges
+
+* **Additional Fields**: Our project manages additional fields such as Telegram handle, module, and contact type. Integrating these fields required modifying both data structures and UI components, as well as adapting backend logic to manage and validate these new attributes.
+* **Enhanced Search Functionality**: We implemented additional search functionalities, allowing users to find contacts by tags and Telegram handles, in addition to names. This required integration of the find command with additional fields.
+* **UI Enhancements**: Significant effort went into redesigning the user interface to make it aesthetically pleasing. In addition to improving layout and visual styling, we introduced a light and dark mode, ensuring each element was clearly visible in both themes.
```
Comment (SlothyCat) [inline docs/DeveloperGuide.md]: Would "we introduced both light and dark modes" sound better? 🤔

**Review by SlothyCat — CHANGES_REQUESTED** [review]
LGTM other than small nitpick with language!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (kimberlytmq) [reply docs/DeveloperGuide.md]: Changed!

**Review by SlothyCat — APPROVED** [review]
LGTM! 😄

**Review by achl1012 — APPROVED** [review]
LGTM! 👍

**Pull request merged**